devito/types/basic.py

```diff
@@ -729,7 +729,7 @@ def dimensions(self):
     def _eval_deriv(self):
         return self
 
-    @cached_property
+    @property
```

Comment:
this is **extremely** scary

Reply:
as discussed offline (slack) this is necessary due to our caching infrastructure which assigns `__dict__` (for performance) rather than copying it, thus preventing us from selectively dropping certain entries from the cached init.